connectors/agent/protocol.py

```diff
@@ -0,0 +1,72 @@
+from elastic_agent_client.generated import elastic_agent_client_pb2 as proto
+from elastic_agent_client.handler.action import BaseActionHandler
+from elastic_agent_client.handler.checkin import BaseCheckinHandler
+from elastic_agent_client.util.logger import logger
+
+
+class ConnectorActionHandler(BaseActionHandler):
+    """Class handling Agent actions.
+
+    As there are no actions that we can respond to, we don't actually do anything here.
+    """
+
+    async def handle_action(self, action: proto.ActionRequest):
+        """Implementation of BaseActionHandler.handle_action
+
+        Right now does nothing as Connectors Service has no actions to respond to.
+        """
+        msg = (
+            f"This connector component can't handle action requests. Received: {action}"
+        )
+        raise NotImplementedError(msg)
+
+
+class ConnectorCheckinHandler(BaseCheckinHandler):
+    """Class handling to Agent check-in events.
+
+    Agent sends check-in events from time to time that might contain
+    information that's needed to run Connectors Service.
+
+    This class reads the events, sees if there's a reported change to connector-specific settings,
+    tries to update the configuration and, if the configuration is updated, restarts the Connectors Service.
+    """
+
+    def __init__(self, client, agent_connectors_config_wrapper, service_manager):
+        """Inits the class.
+
+        Initing this class should not produce side-effects.
+        """
+        super().__init__(client)
+        self.agent_connectors_config_wrapper = agent_connectors_config_wrapper
+        self.service_manager = service_manager
+
+    async def apply_from_client(self):
```

Comment:
This looks a lot like the "Fake example", but I'm worried we'll need to implement more here. This seems to only look at the first unit's `config`, but doesn't consider `log_level`, `features`, or anything else from the checkin service's `sync_component` or `sync_units` method.